### PR TITLE
#169257396  Signin and Signup validations

### PR DESCRIPTION
**Pull request by Mastel22**
#### What does this PR do?
Creates validation schemas for signin and signup
#### Description of Task to be completed?
Added:

- [X] validation schemas
- [X] validation schema middleware
#### How should this be manually tested?

- Git clone this repo
- Run `npm i`
- Run `npm run watch`
- Use Postman 

#### Any background context you want to provide?
-N/A
#### What are the relevant pivotal tracker stories?
[#169257396](https://www.pivotaltracker.com/story/show/169257396)
#### Screenshots

image

image (1)

#### Questions:
None

### CODE REVIEW

**Review by RaymondMwaura — COMMENTED** [review]
Maybe you can move the validation file to the validation folder as shown on the sample project structure below:
https://github.com/andela/learning-facilitators-bn-backend/tree/master/src

**Comment by timolinn** [comment]
> #### How should this be manually tested?
> * Git clone this repo
> 
Please specify how this should be tested.

File: src/middlewares/user.validation.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import db from '../models';
+import Responses from '../utils/response';
+
+// eslint-disable-next-line require-jsdoc
+const findByEmail = async (req, res, next) => {
```
Comment (timolinn) [inline src/middlewares/user.validation.js]: Good job here Mastel! I feel we could have here, a name that is more **reflective** of what this function is doing?

**Review by timolinn — CHANGES_REQUESTED** [review]

File: src/middlewares/user.validation.js (hunk 1 above)
Comment (timolinn) [reply src/middlewares/user.validation.js]: Also add JSDoc here. There's probably an extension in your IDE that could help with that.

File: src/middlewares/validation.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          status: 400,
+          error: error.details[0].message,
+        });
+      }
+
+      req.body = data;
+      next();
+    });
+  }
+  res.status(405).json({
```
Comment (timolinn) [inline src/middlewares/validation.js]: You can use the Response handler for this. Check #18

Comment (Mastel22) [reply src/middlewares/validation.js]: Done now

File: src/middlewares/user.validation.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import db from '../models';
+import Responses from '../utils/response';
+
+/**
+ * Finds if the email passed in the body already exists
```
Comment (timolinn) [inline src/middlewares/user.validation.js]: Change `Finds` to `Checks`.

**Review by timolinn — CHANGES_REQUESTED** [review]

File: src/middlewares/user.validation.js (hunk 3 above)
Comment (Mastel22) [reply src/middlewares/user.validation.js]: Done.

**Review by b0nbon1 — DISMISSED** [review]
Great work

**Review by nshaphan — DISMISSED** [review]
LGTM

**Review by timolinn — COMMENTED** [review]
Well done Mastel, it all looks good to me except that we cannot really test this because the CreateUser endpoint is still non-functional.

**Comment by Mastel22** [comment]
> Well done Mastel, it all looks good to me except that we cannot really test this because the CreateUser endpoint is still non-functional.

**Review by timolinn — DISMISSED** [review]

**Review by RaymondMwaura — CHANGES_REQUESTED** [review]
This works as expected. However, I think you can have it return all the errors in an array form so that the user knows all the errors that they have to fix in one response.

**Comment by Mastel22** [comment]
> This works as expected. However, I think you can have it return all the errors in an array form so that the user knows all the errors that they have to fix in one response.

Sorry it took a while to respond because of the restrictions, but it's resolved now

**Review by b0nbon1 — DISMISSED** [review]
Tested and I have confirmed its working. Great job.
I think it is ready to be merged

**Review by nshaphan — DISMISSED** [review]

**Review by timolinn — DISMISSED** [review]

**Comment by timolinn** [comment]
Good job Mastel, please fix conflicts.

**Review by RaymondMwaura — COMMENTED** [review]
This is well done. Fix conflicts and we'll be good to go.

**Review by b0nbon1 — APPROVED** [review]

**Comment by b0nbon1** [comment]
Great work

**Review by nshaphan — APPROVED** [review]
Looks good!

**Review by RaymondMwaura — APPROVED** [review]
The conflicts are resolved. Good work.

**Pull request merged**